core/crypto/bls/impl/bls_provider_impl.cpp

```diff
@@ -86,6 +86,24 @@ namespace fc::crypto::bls {
               digest.begin());
     return digest;
   }
+
+  outcome::result<Signature> BlsProviderImpl::aggregateSignatures(
+      const std::vector<Signature> &signatures) const {
+    Signature signature;
+    const uint8_t *flat_bytes =
+        reinterpret_cast<const uint8_t *>(signatures.data());
+    size_t flat_size = sizeof(Signature) * signatures.size();
+    AggregateResponse *response = ::aggregate(flat_bytes, flat_size);
+    if (response == nullptr) {
+      return Errors::InternalError;
+    }
+    auto free_response =
+        gsl::finally([response]() { destroy_aggregate_response(response); });
```

Comment:
doesn't clang-tidy blames it? I mean raw pointer passing to lambda.

Reply:
We decided to use raw pointers + gsl::finally instead unique_ptr with custom destructor, and clang-tidy CI tests completes successfully.